src/rpc/mining.cpp

```diff
@@ -544,6 +544,7 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
         pindexPrev = pindexPrevNew;
     }
     CBlock* pblock = &pblocktemplate->block; // pointer for convenience
+    assert(pblock);
```

Comment:
The dereference I'm trying to avoid is not on the line above, but a couple of lines below - [see here](https://github.com/bitcoin/bitcoin/blob/master/src/rpc/mining.cpp#L551) (`pblock->nNonce = 0;`).

Is your reading that `if`-branch on [lines 526-546](https://github.com/bitcoin/bitcoin/blob/master/src/rpc/mining.cpp#L526-L545) will always be taken?

If that branch is not taken then `&pblocktemplate->block` could theoretically be `NULL`, right? :-)